taqueria-vscode-extension-web-ui/src/data-editors/MichelineEditor.tsx

```diff
@@ -0,0 +1,236 @@
+import { emitMicheline, Parser } from '@taquito/michel-codec';
+import React, { useState } from 'react';
+import { WebviewApi } from 'vscode-webview';
+import { MichelineEditorMessageHandler } from '../interopTypes';
+import { DataEditorNode } from './DataEditorNode';
+import './MichelineEditor.css';
+import { MichelineValue as ValidationMichelineValue, validate } from './MichelineValidator';
+import { ValidationResultDisplay } from './ValidationResultDisplay';
+import { VSCodeButton } from './VsCodeWebViewUIToolkitWrappers';
+
+const parser = new Parser();
+
+export type OriginalFormat = 'micheline' | 'json';
+
+export type MichelineValue = {
+	originalFormat: 'micheline';
+	value: string;
+} | {
+	originalFormat: 'json';
+	value: object;
+};
+
+const json2Micheline = (v: object) => {
+	let michelson: string | undefined = undefined;
+	try {
+		const expression = parser.parseJSON(v);
+		michelson = emitMicheline(expression);
+		return michelson;
+	} catch (e: any) {
+		return `${e}`;
+	}
+};
+
+const micheline2Json = (v: string) => {
+	let json: object | undefined | null = undefined;
+	try {
+		json = parser.parseMichelineExpression(v);
+		return json;
+	} catch (e: any) {
```

Comment:
Should we catch and log these errors?

Reply:
No. These happen all the time, for instance when an int field is still empty, or many other reasons.